builder/builder.go

```diff
@@ -132,11 +144,19 @@ func (b *builder) setOptions(opts []Option) {
 	}
 }
 
-func (b *builder) buildParser(g *ast.Grammar) error {
-	b.writeInit(g.Init)
-	b.writeGrammar(g)
+func (b *builder) buildParser(grammar *ast.Grammar) error {
+	haveLeftRecursion, err := PrepareGrammar(grammar)
```

Comment:
No, because if the flag is not set, then you still need to check for recursion and return an error, as this leads to a hang.

https://github.com/mna/pigeon/issues/79

Reply:
Yes, this makes sense.